floodwatch/src/js/components/Chart.js

```diff
@@ -0,0 +1,271 @@
+// @flow
+
+import React, { Component } from 'react';
+import { Row, Col } from 'react-bootstrap';
+import * as d3 from 'd3';
+import _ from 'lodash';
+import colors from './colors'
+
+type Props = {
+  barData: Array;
+  currentTopic: string;
+  side: string;
+  updateMouseOver: Function;
+}
+
+type State = {
+  height: Number;
+}
+
+function initialState() {
+  return {
+    height:500
+  }
+}
+
+export class Chart extends React.Component {
+  state: State;
+  props: Props
+
+  constructor(props: Props) {
+    super(props);
+    this.state = initialState();
+  }
+
+  componentDidMount() {
+    let ctx = this;
+    let svg = d3.select('.svg-' + this.props.side).append('svg').attr('width', '100%').attr('height', ctx.state.height)
+    let defs = svg.append('defs')
+
+    let keys = Object.keys(colors)
+    keys.map((key) => {
+      console.log(key)
+
+      let thisGradient = defs.append('linearGradient')
+        .attr('id', key)
+        .attr('x1', '0%')
+        .attr('x2', '100%')
+        .attr('y1', '0%')
+        .attr('y2', '100%');
+
+      thisGradient.append('stop')
+        .attr('class', 'start')
+        .attr('offset', '0%')
+        .attr('stop-color', colors[key][1])
+        .attr('stop-opacity', 1);
+
+      thisGradient.append('stop')
+        .attr('class', 'end')
+        .attr('offset', '90%')
+        .attr('stop-color', colors[key][0])
+        .attr('stop-opacity', 1);
+
+    })
+
+    this.drawRects(svg, this.props.barData);
+    this.setState({
+      svg: svg,
+    })
+  }
+
+  drawRects(svg, mydata) {
+    // Some of this  d3is redundant, but it's tricky to strip out before testing it with the query changer. Willfix.
+
+    let data = _.cloneDeep(mydata)
+    let ctx = this;
```

Comment:
This looks like d3 boilerplate - you don't have to do this anymore if you use arrow functions. Maybe hold off on fixing it until you do whatever d3 cleanup you have in mind.